### PR TITLE
[master] Enable executor properties in salt-ssh

### PR DESCRIPTION
**Pull request by m-czernek**
### What does this PR do?

The `salt-ssh` binary does not accept the `module_executors` and `executor_opts` properties, yet the `salt` binary does. This PR unifies the differences so that we can use the executors property for `salt-ssh` as well.

### Previous Behavior
`salt-ssh` could only read the `module_executors` property from the roster file (`minion_opts`) or from the `ssh_minion_opts` config.

### New Behavior
`salt-ssh` can read the `module_executors` property from the command-line arguments as well.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by Akm0d — DISMISSED** [review]
LGTM but needs a test

**Review by twangboy — CHANGES_REQUESTED** [review]
Please add a changelog and a test for this.

**Comment by twangboy** [comment]
Changelog and test please?

File: salt/client/ssh/__init__.py [hunk 1]
```diff
@@ -1238,6 +1238,10 @@ def __init__(
         self.minion_opts.update(opts.get("ssh_minion_opts", {}))
         if minion_opts is not None:
             self.minion_opts.update(minion_opts)
+        if opts.get("module_executors") is not None:
+            self.minion_opts.update({"module_executors": opts.get("module_executors")})
+        if opts.get("executor_opts") is not None:
+            self.minion_opts.update({"executor_opts": opts.get("executor_opts")})
```
Comment (twangboy) [inline salt/client/ssh/__init__.py:1244]: Switch from .update() to .setdefault() so that the global master/CLI options are only applied as a fallback when they aren't already explicitly declared on the minion:

```python
        # Fall back to global master config/CLI options if not explicitly 
        # defined for this target minion in the roster.
        if opts.get("module_executors") is not None:
            self.minion_opts.setdefault("module_executors", opts.get("module_executors"))
        if opts.get("executor_opts") is not None:
            self.minion_opts.setdefault("executor_opts", opts.get("executor_opts"))
```

Additionally, the new unit test test_executor_opts_propagate_to_minion_opts passes because the test fixture cleanly strips these keys out of the base options beforehand, which masks this clobbering edge case.

Please write an additional test case that ensures an existing module_executors key inside ssh_minion_opts isn't overridden by the master config.

**Review by twangboy — CHANGES_REQUESTED** [review]

**Comment by m-czernek** [comment]
This feature work got priority removed since the PR creation, closing this PR for now.
